packages/suite/src/components/suite/WelcomeLayout/index.tsx

```diff
@@ -84,50 +89,105 @@ const SettingsWrapper = styled.div`
     align-self: flex-end;
 `;
 
-interface Props {
-    children: React.ReactNode;
-}
+// welcome bar should take two-fifths of screen (minus padding) unless screen is too narrow/wide
+const getWelcomeBarWidth = (width: number | null) => {
+    if (width! * 0.4 < 380) return '380px';
+    if (width! * 0.4 > 660) return '660px';
+
+    return 'calc(40vw - 16px)';
+};
 
 // WelcomeLayout is a top-level wrapper similar to @suite-components/SuiteLayout
 // used in Preloader and Onboarding
-const WelcomeLayout = ({ children }: Props) => {
+const WelcomeLayout: React.FC = ({ children }) => {
     const { banner } = useMessageSystem();
+    const { guideOpen, screenWidth } = useSelector(state => ({
+        guideOpen: state.guide.open,
+        screenWidth: state.resize.screenWidth,
+    }));
+
+    // do not animate welcome bar on initial load
+    const [firstRenderDone, setFirstRenderDone] = useState(false);
+    useEffect(() => {
+        setFirstRenderDone(true);
+    }, []);
+    const [welcomeBarWidth, setWelcomeBarWidth] = useState(getWelcomeBarWidth(screenWidth));
+    useEffect(() => {
+        setWelcomeBarWidth(getWelcomeBarWidth(screenWidth));
+    }, [screenWidth]);
+
     return (
         <Wrapper>
             {banner && (
                 <BannerWrapper>
                     <MessageSystemBanner message={banner} />
                 </BannerWrapper>
             )}
-            <Welcome>
-                <Expander>
-                    <TrezorLogo type="suite" width="128px" />
-                    <WelcomeTitle>
-                        <Translation id="TR_ONBOARDING_WELCOME_HEADING" />
-                    </WelcomeTitle>
-                </Expander>
-                <Bottom>
-                    {isWeb() && (
-                        <StyledTrezorLink size="small" variant="nostyle" href={SUITE_URL}>
-                            <Button variant="tertiary" icon="EXTERNAL_LINK" alignIcon="right">
-                                <Translation id="TR_ONBOARDING_DOWNLOAD_DESKTOP_APP" />
-                            </Button>
-                        </StyledTrezorLink>
+            <WelcomeWrapper>
+                <AnimatePresence>
+                    {!guideOpen && (
+                        <MotionWelcome
+                            initial={
+                                !firstRenderDone
+                                    ? {
+                                          width: welcomeBarWidth,
+                                      }
+                                    : { width: 0 }
+                            }
```

Comment:
I believe that framer-motion could handle it without custom `firstRenderDone` state, but not sure.

Have you tried `initial={false}`? https://www.framer.com/docs/animation/##mount-animations

Reply:
yes, I have. If you use `initial={false}`, guide opening animation is missing

Reply:
👌